src/adapters/github.ts

```diff
@@ -74,13 +79,51 @@ class GithubAdapter extends GitAdapter {
   }
 
   public reposEqual(repo1: IRepo, repo2: IRepo): boolean {
-    return isEqual(repo1, repo2);
+    return repo1.owner === repo2.owner && repo1.name === repo2.name;
   }
 
   public stringifyRepo({ owner, name }: IRepo): string {
     return `${owner}/${name}`;
   }
 
+  public async resetRepoBeforeApply(repo: IRepo, force: boolean) {
+    const { defaultBranch } = repo;
+    // First, get any changes from the remote
+    // --prune will ensure that any remote branch deletes are reflected here
+    await this.git(repo).fetch(['origin', '--prune']);
+
+    if (!force) {
+      const safetyStatus = await this.checkActionSafety(repo);
+      if (safetyStatus === SafetyStatus.PullRequestExisted) {
+        throw new Error('Remote branch did not exist, but a pull request does or did; try with --force-reset-branch?');
+      } else if (safetyStatus === SafetyStatus.NonShepherdCommits) {
+        throw new Error('Found non-Shepherd commits on remote branch; try with --force-reset-branch?');
+      }
+    }
+
+    // If we got this far, we can go ahead and reset to the default branch
+    await this.git(repo).reset(['--hard', `origin/${defaultBranch}`]);
+  }
```

Comment:
I don't know if it's because I'm familiar with this flow now after the past few discussions, but this logic was a breeze to read through and understand! 👍